cedar-policy-validator/src/schema/namespace_def.rs

```diff
@@ -64,80 +64,131 @@ pub(crate) fn is_action_entity_type(ty: &Name) -> bool {
+    /// in this schema fragment).
     pub(super) attributes: WithUnresolvedTypeDefs<Type>,
-    /// The direct parent entity types for this entity type come from the
-    /// `memberOfTypes` list. These types might be declared in a different
-    /// namespace, so we will check if they are declared in any fragment when
-    /// constructing a `ValidatorSchema`.
+    /// Direct parent entity types for this entity type.
+    /// These are fully qualified `Name`s, but may be entity types declared in a
+    /// different namespace or schema fragment.
+    /// We will check for undeclared parent types when combining fragments into
+    /// a `ValidatorSchema`.
     pub(super) parents: HashSet<Name>,
 }
 
-/// Action declarations held in a `ValidatorNamespaceDef`. Entity types
-/// referenced here do not need to be declared in the schema.
+/// Holds a map from (fully qualified) `EntityUID`s of action definitions
+/// to their corresponding `ActionFragment`. The action `EntityUID`s (keys in the map)
+/// are fully qualified, and inside the `ActionFragment`s (values in the map),
+/// all entity/common type references (including references to other actions)
+/// are also fully qualified.
+///
+/// However, the `ActionFragment`s may reference undeclared (but fully
+/// qualified) entity/common types and actions.
 #[derive(Debug)]
 pub struct ActionsDef {
     pub(super) actions: HashMap<EntityUID, ActionFragment>,
 }
 
+/// Holds the information about an action that comprises an action definition.
+///
+/// In this representation, references to common types may not yet have been
+/// fully resolved/inlined. But, all entity/common type references (including
+/// references to other actions) are fully qualified. This `ActionFragment` may
+/// reference undeclared (but fully qualified) entity/common types and actions.
 #[derive(Debug)]
 pub struct ActionFragment {
     /// The type of the context record for this action. The type is wrapped in
     /// a `WithUnresolvedTypeDefs` because it may refer to common types which
-    /// are not defined in this schema fragment.
+    /// have not yet been resolved/inlined (e.g., because they are not defined
+    /// in this schema fragment).
     pub(super) context: WithUnresolvedTypeDefs<Type>,
     /// The principals and resources that an action can be applied to.
     pub(super) applies_to: ValidatorApplySpec,
     /// The direct parent action entities for this action.
+    /// These are fully qualified `EntityUID`s, but may be actions declared in a
+    /// different namespace or schema fragment, and thus not declared yet.
+    /// We will check for undeclared parents when combining fragments into a
+    /// `ValidatorSchema`.
     pub(super) parents: HashSet<EntityUID>,
     /// The types for the attributes defined for this actions entity.
+    /// Although these are not literally wrapped in `WithUnresolvedTypeDefs`,
+    /// they are in spirit: they may refer to common types which have not yet
```

Comment:
I don't think this is accurate. These should be fully specified.

(we should honestly just rip out action attributes, they're not in any public api)

Reply:
Fixed the comment. Thanks for the review.